### ISSUE TITLE
hooks not being called when on test (aka Using Nuxt.js Programmatically)

### ISSUE THREAD

**Issue by volkovmqx** [nuxt/nuxt.js#4881]
### Version

[v2.4.0](https://github.com/nuxt.js/releases/tag/v2.4.0)

### Reproduction link

[https://codesandbox.io/s/r0p4l9j85q](https://codesandbox.io/s/r0p4l9j85q)

### Steps to reproduce

basically use a hook and use a test (or use the nuxt.js API: new nuxt() and build())

### What is expected ?

when using
```
    nuxt = new Nuxt(config)
    await new Builder(nuxt).build()
```
hooks should be called

### What is actually happening?

the build hooks not being called (and possibly other hooks)

### Additional comments?

basically i am doing some copy paste from different paths into the src folder before build. so i am combining different pages/components into a single project. the build works fine. but the test are not.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8561">#c8561</a>)</em></sub></div>

**Comment by pimlie** [comment]
Your test is not using `nuxt.config.js` but only the config you define in the test. You need to manually load the `nuxt.config.js`, change rootDir/dev if you need to as you do now and pass that to Nuxt. Check the loadFixture method which the Nuxt tests use

**Comment by volkovmqx** [comment]
very well, i thought i had that. now that raise another issue, i believe it's related to lodash transpiling, you can check the same link. i've updated the demo. 
however, if i pass an object with the hooks to the nuxt constructor, it gets executed which is a great step !!

**Comment by manniL** [comment]
@volkovmqx probably, yes. Using ava with `esm` might be your solution (or writing your nuxt config in CJS syntax).

As it's no Nuxt bug, I'll close the issue ☺️ 
Thanks @pimlie for helping!
